src/chainstate/stacks/index/proofs.rs

```diff
@@ -563,10 +556,10 @@ impl TrieMerkleProof {
     /// Given a segment proof, the deepest node's hash, and the hash of the trie root, verify that
     /// the segment proof is well-formed.
     /// If so, calculate the root hash of the segment and return it.
-    fn verify_segment_proof(proof: &[TrieMerkleProofType], node_hash: &TrieHash) -> Option<TrieHash> {
+    fn verify_segment_proof(proof: &[TrieMerkleProofType], node_hash: &TrieHash, block_map: &BlockHashMap) -> Option<TrieHash> {
```

Comment:
I'm not sure how I feel about the use of the `block_map` here.  The verifier is supposed to be called by light clients.  Light clients don't have access to the chain state; only Bitcoin and Stacks headers.  Can the block map be built from just the block headers?

Reply:
Yeah— this should change. The Merkle proof should replace the u32s with the block headers in place. Using the block map was temporary for now, because I wanted to make sure the proof verification code worked like this before modifying the proofs structure.

Reply:
Addressed in da7f5c2